src/conn/conn_api.c

```diff
@@ -3044,8 +3044,13 @@ wiredtiger_open(const char *home, WT_EVENT_HANDLER *event_handler, const char *c
         WT_ERR_MSG(
           session, EINVAL, "direct I/O configuration is incompatible with mmap_all configuration");
 
-    WT_ERR(__wt_config_gets(session, cfg, "prefetch", &cval));
+    WT_ERR(__wt_config_gets(session, cfg, "prefetch.available", &cval));
+    conn->prefetch_available = cval.val != 0;
+    WT_ERR(__wt_config_gets(session, cfg, "prefetch.enabled", &cval));
     conn->prefetch_auto_on = cval.val != 0;
+    if (conn->prefetch_auto_on && !conn->prefetch_available)
+        WT_ERR_MSG(
+          session, EINVAL, "pre-fetching cannot be enabled if pre-fetching server is unavailable");
```

Comment:
I'd tie this error message to the config a bit more closely - the config makes no mention of any server. Perhaps "pre-fetching cannot be enabled by default on new sessions if pre-fetching is configured as available"?